### DISCUSSION TITLE
How to access $values in a json variable to set as items for a ForEach activity?

### DISCUSSION THREAD

**Discussion by MaxBrooks114** [elsa-workflows/elsa-core#4097] [Q&A]
I have this json object saved to a variable in elsa workflow designer. The variable was created with a setvariable activity which set this part of the body of an http request as a variable called "users"

```
{
  "users": {
    "$type": "System.Collections.Generic.List`1[[System.Object, System.Private.CoreLib]], System.Private.CoreLib",
    "$values": [
      {
        "$id": "3",
        "$type": "System.Dynamic.ExpandoObject, System.Linq.Expressions",
        "UsersID": 2,
        "AssociatedUsersID": 51,
        "UserEmail": "bob@Site.com",
        "Title": "Manager",
        "UserFirstName": "Bob",
        "UserLastName": "Loblaw",
        "NotificationTypes": {
          "$type": "System.Collections.Generic.List`1[[System.Object, System.Private.CoreLib]], System.Private.CoreLib",
          "$values": [
            "Vendor Assigned"
          ]
        }
      }
    ]
  }
}
```
I am trying to access the "$values" collection in the object for the "ForEach" activity in the designer, but items is always null. Here's what I've tried so far:

getVariable('users')
getVariable('users').$values
getVariable('users')["$values"]
getVariable('users')['$values']

 What am I doing wrong here?

**Comment by sfmskywalker** [comment]
The `users` field should have been deserialized into a strongly typed C# array of objects, but evidently this doesn't happen, since `getVariable('users') doesn't work. This could be a bug. What version are you using?
